File: Baekjoon/Class1/35.8958.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n;
    cin >> n;
    while (n--) {
        int score = 0, total = 0;
        string input;
        cin >> input;
        for (int i = 0; i < input.length(); i++)
            if (input[i] == 'O')
                total += ++score;
            else
                score = 0;
        cout << total << '\n';
    }
}

File: Baekjoon/Class1/35.8958.py
import sys
input = sys.stdin.readline

n = int(input())
for _ in range(n):
    score = 0
    total = 0
    string = input().rstrip()
    for ch in string:
        if ch == "O":
            score += 1
            total += score
        else:
            score = 0
    print(total)